ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -358,27 +371,329 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
+            auto const& w = ip_data.integration_weight;
+
+            double C_int_pt = 0.0;
+            double p_int_pt = 0.0;
+
+            NumLib::shapeFunctionInterpolate(local_C, N, C_int_pt);
+            NumLib::shapeFunctionInterpolate(local_p, N, p_int_pt);
+
+            // porosity model
+            auto const porosity =
+                _process_data.porous_media_properties.getPorosity(t, pos)
+                    .getValue(t, pos, 0.0, C_int_pt);
+
+            // Use the fluid density model to compute the density
+            // TODO: Concentration of which component as one of arguments for
+            // calculation of fluid density
+            vars[static_cast<int>(
+                MaterialLib::Fluid::PropertyVariableType::C)] = C_int_pt;
+            vars[static_cast<int>(
+                MaterialLib::Fluid::PropertyVariableType::p)] = p_int_pt;
+            auto const density = _process_data.fluid_properties->getValue(
+                MaterialLib::Fluid::FluidPropertyType::Density, vars);
+
+            auto const& K =
+                _process_data.porous_media_properties.getIntrinsicPermeability(
+                    t, pos).getValue(t, pos, 0.0, 0.0);
+            // Use the viscosity model to compute the viscosity
+            auto const mu = _process_data.fluid_properties->getValue(
+                MaterialLib::Fluid::FluidPropertyType::Viscosity, vars);
+
+            GlobalDimMatrixType const K_over_mu = K / mu;
+
+            const double drho_dp = _process_data.fluid_properties->getdValue(
+                MaterialLib::Fluid::FluidPropertyType::Density,
+                vars,
+                MaterialLib::Fluid::PropertyVariableType::p);
+            const double drho_dC = _process_data.fluid_properties->getdValue(
+                MaterialLib::Fluid::FluidPropertyType::Density,
+                vars,
+                MaterialLib::Fluid::PropertyVariableType::C);
+
+            // matrix assembly
+            local_M.noalias() += w * N.transpose() * porosity * drho_dp * N;
+            local_K.noalias() +=
+                w * dNdx.transpose() * density * K_over_mu * dNdx;
+
+            if (_process_data.has_gravity)
+                local_b.noalias() +=
+                    w * density * density * dNdx.transpose() * K_over_mu * b;
+
+            // coupling term
+            {
+                double C0_int_pt = 0.0;
+                NumLib::shapeFunctionInterpolate(local_C0, N, C0_int_pt);
+
+                local_b.noalias() -= w * N.transpose() * porosity * drho_dC *
+                                     (C_int_pt - C0_int_pt) / dt;
+            }
+        }
+    }
+
+    void assembleComponentTransportEquation(
+        double const t, std::vector<double>& local_M_data,
+        std::vector<double>& local_K_data,
+        std::vector<double>& /*local_b_data*/,
+        LocalCoupledSolutions const& coupled_xs)
+    {
+        auto const transport_process_id = coupled_xs.process_id;
```

Comment:
Is this the same `transport_process_id` as in the above method?

Reply:
Not the same.

Reply:
So both variables have the same name, but are used for different things. Which one is the 'real' transport process id?

Reply:
Fixed.